chimney/src/test/scala/io/scalaland/chimney/PartialTransformerProductSpec.scala

```diff
@@ -0,0 +1,1106 @@
+          .intoPartial[Person]
+          .withFieldComputedPartial(
+            _.name,
+            pf =>
+              if (pf.name.isEmpty) partial.Result.fromEmpty
+              else partial.Result.fromValue(pf.name.toUpperCase())
+          )
+          .withFieldComputed(_.age, _.age.toInt) // must catch exceptions
+          .withFieldComputedPartial(
+            _.height,
+            pf => partial.Result.fromOption(pf.height.parseDouble)
+          )
+          .transform
+
+        result.asOption ==> Some(expected)
+        result.asEither ==> Right(expected)
+        result.asErrorPathMessagesStrings ==> Iterable.empty
+      }
+
+      test("failure with error handling") {
+        val invalidForm = PersonForm("", "foo", "bar")
+
+        val result = invalidForm
+          .intoPartial[Person]
+          .withFieldComputedPartial(
+            _.name,
+            pf =>
+              if (pf.name.isEmpty) partial.Result.fromEmpty
+              else partial.Result.fromValue(pf.name.toUpperCase())
+          )
+          .withFieldComputed(_.age, _.age.toInt) // must catch exceptions
+          .withFieldComputedPartial(
+            _.height,
+            pf => partial.Result.fromOption(pf.height.parseDouble)
+          )
+          .transform
+
+        result.asOption ==> None
+        result.asErrorPathMessagesStrings ==> Iterable(
+          "name" -> "empty value",
+          "age" -> "For input string: \"foo\"",
+          "height" -> "empty value"
+        )
+      }
+    }
+
+    test("recursive partial transform with nested validation") {
+
+      import trip._
+
+      implicit val personPartialTransformer: PartialTransformer[PersonForm, Person] =
+        Transformer
+          .definePartial[PersonForm, Person]
+          .withFieldComputedPartial(_.age, _.age.parseInt.toPartialTransformerResultOrString("bad age value"))
+          .withFieldComputedPartial(
+            _.height,
+            _.height.parseDouble.toPartialTransformerResultOrString("bad height value")
+          )
+          .buildTransformer
+
+      test("success") {
+
+        val okTripForm = TripForm("100", List(PersonForm("John", "10", "140"), PersonForm("Caroline", "12", "155")))
+
+        val result = okTripForm
+          .intoPartial[Trip]
+          .withFieldComputedPartial(_.id, _.tripId.parseInt.toPartialTransformerResultOrString("bad trip id"))
+          .transform
+
+        result.asOption ==> Some(Trip(100, Vector(Person("John", 10, 140), Person("Caroline", 12, 155))))
+      }
+
+      test("failure with error handling") {
+
+        val badTripForm =
+          TripForm("100xyz", List(PersonForm("John", "10", "foo"), PersonForm("Caroline", "bar", "155")))
+
+        val result = badTripForm
+          .intoPartial[Trip]
+          .withFieldComputedPartial(_.id, _.tripId.parseInt.toPartialTransformerResultOrString("bad trip id"))
+          .transform
+
+        result.asOption ==> None
+        result.asEither ==> Left(
```

Comment:
I'm unsure about constructing these path-prepended error values in tests. They're pretty bad to write and read.

As a middleground, how about using just `asErrorPathMessages` to verify the errors? (look on the rebased branch)